compiler/codegen/codegen.cpp

```diff
@@ -1117,12 +1117,17 @@ static void codegen_header_compilation_config() {
   if (cfgfile.fptr != NULL) {
     FILE* save_cfile = gGenInfo->cfile;
 
-    gGenInfo->cfile = cfgfile.fptr;
+    fprintf(cfgfile.fptr, "/*** %s ***/\n", "Compilation Info");
+    fprintf(cfgfile.fptr, "\n#include <stdio.h>");
+    fprintf(cfgfile.fptr, "\n#include \"chpltypes.h\"\n\n");
 
-    genComment("Compilation Info");
-
-    fprintf(cfgfile.fptr, "\n#include <stdio.h>\n");
-    fprintf(cfgfile.fptr, "\n#include \"chpltypes.h\"\n");
+    if (llvmCodegen) {
+#ifdef HAVE_LLVM
+      gGenInfo->cfile = NULL;
+#endif
+    } else {
+      gGenInfo->cfile = cfgfile.fptr;
+    }
```

Comment:
If you configure Chapel with a launcher (e.g. by doing `export CHPL_COMM=gasnet` and running `make`), does this still generate the .c file for use by the launcher, even in `--llvm` mode? Please check. I was expecting we'd need to add a function for codegen_header_compilation_config to call to do both LLVM and C in that configuration.